prov/util/src/util_shm.c

```diff
@@ -145,6 +145,35 @@ size_t smr_calculate_size_offsets(size_t tx_count, size_t rx_count,
 	return total_size;
 }
 
+static int smr_retry_map(const char *name, int *fd)
+{
+	char tmp[NAME_MAX];
+	struct smr_region *old_shm;
+	struct stat sts;
+	int shm_pid;
+
+	*fd = shm_open(name, O_RDWR | O_CREAT, S_IRUSR | S_IWUSR);
+	if (*fd < 0)
+		return -errno;
+
+	old_shm = mmap(NULL, sizeof(*old_shm), PROT_READ | PROT_WRITE,
+		       MAP_SHARED, *fd, 0);
+	if (old_shm == MAP_FAILED)
+		return -errno;
+
+	shm_pid = old_shm->pid;
```

Comment:
A magic value in addition to the version number might not be a bad idea either.

Reply:
Can you elaborate on what you mean by a magic value?

Reply:
It's common to put some known value as a sanity check that can be verified that this file was created by the shm provider and not by something else (as unlikely as that is to happen). Otherwise your version check might fail in a weird way.

Reply:
To add to this, a magic value would be 32 or 64 bits, so that the likelihood of it randomly being in memory would be rare.  The version is only 8-bits and is a simple incrementing index.  Plus, magic values allow you to find fun and clever things to spell out in hex.

I think we need the version check.  A magic value could go either way.  The file name provides something of a check along those lines.